module_1/first-react-app/src/components/Movie.js

```diff
@@ -0,0 +1,106 @@
+import { useState, useEffect } from 'react';
+import { UNKNOWN_MOVIE } from '../literals';
+import { NOT_APPLICABLE } from '../literals';
+import Dialog from './Dialog';
+import MovieForm from './MovieForm';
+import  '../css/Movie.css';
+
+const Movie = ({movie , onMovieClick }) => {
+
+    const displayedGenre = movie.genres ? movie.genres.flat().join(", ") : NOT_APPLICABLE;
+    const displayedMovieName = movie.title ? movie.title : UNKNOWN_MOVIE;
+    const displayedReleaseYear = movie.release_date ? movie.release_date.substring(0, 4) : NOT_APPLICABLE;
+
+    const [selectedOption, setSelectedOption] = useState(null);
+    const [isDialogOpen, setIsDialogOpen] = useState(false);
+
+    const handleOptionSelect = (event, option) => {
+        setSelectedOption(option);
+        setIsDialogOpen(true);
+    }
+
+    const handleCloseDialog = (event) => {
+        setIsDialogOpen(false);
+      };
+
+    const renderSelectedComponent = () => {
+        switch (selectedOption) {
+            case 'Edit Movie':
+                return (
+                    <>
+                        {isDialogOpen ? (
+                            <Dialog title="Edit Movie" onClose={(e) => handleCloseDialog(e)}>
+                                <MovieForm initialMovie={movie} onSubmit={null} />
+                            </Dialog>
+                        ) : null}
+                    </>
+                )
+            case 'Delete Movie':
+                return (
+                    <>
+                        {isDialogOpen ? (
+                            <Dialog title="Delete Movie" onClose={(e) => handleCloseDialog(e)}>
+                                <MovieForm initialMovie={movie} deleteMovie={true} onSubmit={null} />
+                            </Dialog>
+                        ) : null}
+                    </>
```

Comment:
we can move that code to the separate component because looks like we use it several times